### ISSUE TITLE
IDEs launched through the extension do not source the terminal configuration

### ISSUE THREAD

**Issue by gpunto** [raycast/extensions#1514]
# Extension – [JetBrains Toolbox Search Recent Projects](https://github.com/raycast/extensions/tree/main/extensions/jetbrains)

Author: @gdsmith 

## Description

If I launch one of my recent projects through the extension, the `.zshrc` file is not sourced. This means, for example, that the IDE won't see the environment variables that I set in `.zshrc`. The configuration is sourced correctly when I launch the IDE from the Toolbox or the terminal.

## Steps To Reproduce

Taking IntelliJ IDEA as an example:

1. Add an environment variable to your `.zhsrc`
2. Open IDEA through the extension
3. Open Preferences > Tools > Terminal and tap on the small icon on the right of Environment Variables
4. Check that the environment variable you added in 1. is not there

If you close IDEA and open it from the Toolbox, the variable should be there.

## The current behavior

The terminal configuration is not sourced.

## The expected behavior

The terminal configuration is sourced.

### Raycast version

Version: 1.33.2

**Comment by rosswintle** [comment]
Can we re-open this please? This is causing me issues where I'm trying to run command-line tools inside PHPStorm when I launch it from Raycast using the Jetbrains extension.

## Steps to reproduce

With PHPStorm v2022.2 installed

1. Open a PHPStorm project that uses phpcs (PHP CodeSniffer - the coding standards tool) using Raycast.
2. Open Preferences in PHPStorm and go to PHP -> Quality Tools -> PHP_CodeSniffer
3. Select "System PHP" in the dropdown and then click the 3 dots to configure the PHP_CodeSniffer path
4. Add a valid path for PHP_CodeSniffer, and click validate. It should say "Can not run PHP_CodeSniffer".
5. You should be able to Apply the settings anyway. So do that.
6. Open a PHP file. This should run PHPCS and you should get a message pop up saying:

``` 
PHP_CodeSniffer
phpcs: env: php: No such file or directory
```

7. Quit PHPStorm.
8. Open PHPStorm normally using Finder, Spotlight, or whatever and re-open the project
9. Repeat steps 2 to 6. In step 4 the validation should happen correctly. In step 6 you should not get the error pop up.

You can also see what is happening from the PHPStorm log if you search for messages containing `shell env`.

To find the log go to Help -> "Show Log in Finder" in PHPStorm, then open `idea.log`.

When launching PHPStorm normally this will log a message containing something like:

```
loading shell env: /bin/zsh -l -i -c '/Users/rosswintle/Library/Application Support/JetBrains/Toolbox/apps/PhpStorm/ch-1/222.3345.135/PhpStorm.app/Contents/bin/printenv' > '/var/folders/tr/vzqv24fn3dvg8qnjwpy2d42r0000gn/T/ij-env-tmp-dir13056768158474175316/ij-shell-env-data.tmp'
```

When launching from Raycast, you will instead see:

```
loading shell env is skipped: IDE has been launched from a terminal (SHLVL=2)
```

I don't know what you are doing differently, but every other way that I have of opening PHPStorm (including the Alfred command palette, the `rcmd` app-launcher app, and my StreamDeck) seems to do this correctly and includes the environment when launching PHPStorm.

## The current behaviour

When opening a PHPStorm project using the Raycast Jetbrains extension, the environment isnot included when the command to open the application is run, and PHP-based command-line tools (such as PHPCS, PHPStan) fail to run correctly from inside the JetBrains application.

Log shows `loading shell env is skipped` when starting PHPStorm

## The expected behaviour

When opening a PHPStorm project using the Raycast Jetbrains extension, the environment should be included when the command to open the application is run, and PHP-based command-line tools (such as PHPCS, PHPStan) should run correctly from inside the JetBrains application.

Log shows `loading shell env: /bin/zsh -l -i -c <files>` when starting PHPStorm

Thanks

**Comment by gdsmith** [comment]
Thanks for the detailed walkthrough @rosswintle. There's a challenge I'm trying to figure out in that I don't have a way to call the CLI tools with the correct environment variables so needing to get some kind of workaround in place to open the app before opening the project. JB not helping in that the toolbox apps in the app folder are not the actual apps you end up launching and they don't seem to respond well to the open command.

Long story short, I'm aware of the issue and trying to find a solution, so don't worry too much that this is closed, but I don't have any timeline for when it will be fixed.
